### ISSUE TITLE
After exporting, creature plugin for godot can't read .json data without copying files from res:// to user://

### ISSUE THREAD

**Issue by MelimeliDev** [godotengine/godot#39309]
Godot version: `3.2.1-stable` and `3.2.2-beta`

OS/device including version: Windows

Issue description:
Everything works fine on editor, but after exporting it seems that the `.json` and `.mdata` can't be read. (Tested on windows `.exe` and web `.html` exports)

The only way to read them is to copy dir from `res://` to `user://` and then reference the files in `user://` instead.

More detailed explanation:
https://github.com/kestrelm/Creature_Godot/issues/16#issuecomment-559658886

TLDR:
![image](https://user-images.githubusercontent.com/14923217/83824097-2d4fc180-a708-11ea-8bae-f17c8cdb767c.png)

**Comment by Calinou** [comment]
Did you specify `*.json` in the non-resource export filters in the Export dialog?

**Comment by Calinou** [comment]
@MelimeliDev Please [fill in the issue template](https://github.com/godotengine/godot/blob/master/.github/ISSUE_TEMPLATE/bug_report.md) when reporting bugs instead of replacing it.

Also, it would be appreciated if you could make a minimal reproduction project. Did you add `*.json, *.mdata` to the non-resource export filter? If so, maybe this is a bug in Godot (but I'm still not sure, as many people have gotten JSON exports to work).

**Comment by MelimeliDev** [comment]
> Did you specify `*.json` in the non-resource export filters in the Export dialog?

Yes, that part is working fine because the .json files has been successfully copied from res:// to user://, they just can't be read/referenced from res://

Godot version has been added, and I don't have problems with reading .json files outside of this plugin. However, it seems that it's also a godot issue because the plugin works fine if files are read from user://

**Comment by MelimeliDev** [comment]
> Also, it would be appreciated if you could upload a minimal reproduction project. 

The godot that needs to be used is different (compiled with the plugin module) so I'm not sure how helpful it'd be, but any of the demo from the plugin github would do:

https://github.com/kestrelm/Creature_Godot/tree/master/creaturegodot3/Godot3Demo

**Comment by bruvzg** [comment]
Since it's C++ plugin it's probably using system file access API to read "*.json" (and `globalize_path()` to get full path). 

If it is, it won't be able to do so, exported files do not exist in the host file system, but stored in the PCK archive.

Unless plugin is using GDNative FileAccess API to access files there's no straightforward way to mitigate this, in any other way than extracting file.

**Comment by bruvzg** [comment]
And looks like it is the case, plugin is using `std::ifstream`: https://github.com/kestrelm/Creature_Godot/blob/master/creaturegodot/CreatureModule.cpp#L657-L661

To read files from the PCK it should use something like (it's probably is not the only place in the plugin that should be changed):

```c++
#include <File.hpp>
...
	Ref<File> file;
	file.instance();
	if (file->open(filename.c_str(), File::READ) == Error::OK) {
		String data = file->get_as_text();
		std::string text(data.utf8().get_data());

		LoadCreatureJSONDataFromString(text, load_data);

		file.close();
	}
```

instead of:

```c++
	std::ifstream read_file;
        read_file.open(filename_in.c_str());
        std::stringstream str_stream;
        str_stream << read_file.rdbuf();
        read_file.close();
        
        LoadCreatureJSONDataFromString(str_stream.str(), load_data);
```

**Comment by MelimeliDev** [comment]
Thank you! I'll see if the plugin owner can fix this bug.

**Comment by bruvzg** [comment]
Related issue: #36781 and PR #37252.

**Comment by MelimeliDev** [comment]
It's solved! Thanks for the help!

**Comment by alperc84** [comment]
> It's solved! Thanks for the help!

Can you point the solution?
